src/overlays/actors/ovl_En_Tsn/z_en_tsn.c

```diff
@@ -28,58 +40,598 @@ const ActorInit En_Tsn_InitVars = {
+                    func_801477B4(globalCtx);
+                    Animation_MorphToLoop(&this->skelAnime, &object_tsn_Anim_0092FC, -10.0f);
+                    this->actionFunc = func_80AE0304;
+                    this->actor.textId = 0;
+                } else {
+                    func_80151938(globalCtx, 0x1093);
+                    Animation_MorphToLoop(&this->skelAnime, &object_tsn_Anim_0092FC, -10.0f);
+                }
+                break;
+
+            case 0x107E:
+                func_801477B4(globalCtx);
+                this->actionFunc = func_80AE0304;
+                this->actor.textId = 0;
+                break;
+        }
+    }
+}
+
+void func_80AE0304(EnTsn* this, GlobalContext* globalCtx) {
+    if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        this->actionFunc = func_80AE0010;
+        this->unk_220 |= 1;
+        if (this->actor.textId == 0) {
+            func_80ADFF84(this, globalCtx);
+        }
+    } else if ((this->actor.xzDistToPlayer < 150.0f) && Player_IsFacingActor(&this->actor, 0x3000, globalCtx)) {
+        func_800B8614(&this->actor, globalCtx, 160.0f);
+        this->unk_220 |= 1;
+    } else {
+        this->unk_220 &= ~1;
+    }
+    if (ENTSN_GET_Z(&this->actor)) {
+        Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 6, 0x1838, 0x64);
+    } else {
+        Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.home.rot.y, 6, 0x1838, 0x64);
+    }
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+}
+
+void func_80AE0418(EnTsn* this, GlobalContext* globalCtx) {
+    if (Actor_TextboxIsClosing(&this->actor, globalCtx)) {
+        func_801518B0(globalCtx, 0x107D, NULL);
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_80AE0460(EnTsn* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        ENTSN_SET_Z(&this->unk_1D8->actor, false);
+        this->actionFunc = func_80AE0418;
+    } else {
+        Actor_PickUp(&this->actor, globalCtx, GI_95, 2000.0f, 1000.0f);
+    }
+}
+
+void func_80AE04C4(EnTsn* this, GlobalContext* globalCtx) {
+    if (Actor_TextboxIsClosing(&this->actor, globalCtx)) {
+        this->actionFunc = func_80AE0C88;
+    }
+}
+
+void func_80AE04FC(EnTsn* this, GlobalContext* globalCtx) {
+    s32 sp24;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (Message_GetState(&globalCtx->msgCtx) == 0x10) {
+        sp24 = func_80123810(globalCtx);
+        if (sp24 != 0) {
+            gSaveContext.weekEventReg[26] |= 2;
+        }
+
+        if (sp24 > 0) {
+            sp24 = sp24;
+            func_801477B4(globalCtx);
+            this->actionFunc = func_80AE0704;
+            if (sp24 == 19) {
+                if (CHECK_QUEST_ITEM(QUEST_UNK_19)) {
+                    if (func_8013A4C4(9)) {
+                        player->actor.textId = 0x107B;
+                        return;
+                    }
+
+                    if (func_8013A4C4(11)) {
```

Comment:
```suggestion
                    if (func_8013A4C4(0xB)) {
```
This is a flag so better to have it in hex